contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/checkstyle.py

```diff
@@ -216,19 +226,34 @@ def _constraints_are_whitelisted(self, constraint_tuple):
     return all(version.parse(constraint) in self._acceptable_interpreter_constraints
            for constraint in constraint_tuple)
 
+  class CheckstyleSetupError(TaskError): pass
+
+  class CheckstyleRunError(TaskError):
+
+    def __init__(self, py3_was_linted, failures_by_min_interpreter, *args, **kwargs):
+      self.py3_was_linted = py3_was_linted
+      self.failures_by_min_interpreter = failures_by_min_interpreter
+      super(Checkstyle.CheckstyleRunError, self).__init__(*args, **kwargs)
+
   def execute(self):
     """"Run Checkstyle on all found non-synthetic source files."""
-    python_tgts = self.context.targets(
-      lambda tgt: isinstance(tgt, (PythonTarget))
-    )
-    if not python_tgts:
-      return 0
-    interpreter_cache = PythonInterpreterCache.global_instance()
-    with self.invalidated(self.get_targets(self._is_checked)) as invalidation_check:
-      failure_count = 0
-      tgts_by_compatibility, _ = interpreter_cache.partition_targets_by_compatibility(
+    if self.skip_execution:
+      return
+
+    if self.act_transitively:
+      all_targets = self.get_targets(self._is_checked)
+    else:
+      all_targets = filter(self._is_checked, self.context.target_roots)
+
+    with self.invalidated(all_targets) as invalidation_check:
+      tgts_by_compatibility, _ = self._interpreter_cache.partition_targets_by_compatibility(
         [vt.target for vt in invalidation_check.invalid_vts]
       )
+
+      failure_count = 0
+      failures_by_min_interpreter = defaultdict(int)
+      py3_was_linted = False
+
       for filters, targets in tgts_by_compatibility.items():
         if self.get_options().interpreter_constraints_whitelist is None and not self._constraints_are_whitelisted(filters):
           deprecated_conditional(
```

Comment:
Can you please pull Chris into this review and have hime help align the deprecated_conditional language below with the now deprecated `--interpreter-constraints-whitelist` option?